plugins/partiql-plugin/src/main/kotlin/org/partiql/plugin/internal/fn/scalar/FnEq.kt

```diff
@@ -34,20 +52,38 @@ import org.partiql.value.PartiQLValueType.STRUCT
 import org.partiql.value.PartiQLValueType.SYMBOL
 import org.partiql.value.PartiQLValueType.TIME
 import org.partiql.value.PartiQLValueType.TIMESTAMP
+import org.partiql.value.SexpValue
+import org.partiql.value.StringValue
+import org.partiql.value.StructValue
+import org.partiql.value.SymbolValue
+import org.partiql.value.TimeValue
+import org.partiql.value.TimestampValue
+import org.partiql.value.boolValue
+import org.partiql.value.check
 
 @OptIn(PartiQLValueExperimental::class, PartiQLFunctionExperimental::class)
 internal object FnEq0 : PartiQLFunction.Scalar {
 
     override val signature = FunctionSignature.Scalar(
         name = "eq",
         returns = BOOL,
-        parameters = listOf(FunctionParameter("lhs", ANY), FunctionParameter("rhs", ANY)),
+        parameters = listOf(
+            FunctionParameter("lhs", ANY),
+            FunctionParameter("rhs", ANY),
+        ),
         isNullCall = true,
         isNullable = false,
     )
 
+    // TODO ANY, ANY equals not clearly defined at the moment.
     override fun invoke(args: Array<PartiQLValue>): PartiQLValue {
-        TODO("Function eq not implemented")
+        val lhs = args[0]
+        val rhs = args[1]
+        return if (lhs.isNull || rhs.isNull) {
+            boolValue(null)
+        } else {
+            boolValue(lhs == rhs)
+        }
```

Comment:
Consider:  `MISSING = a` where a is a non-missing value, I am not sure if this should return false...
At the moment maybe just add a note to this? Or throw a TypeCheckException for consistency?